content/en/docs/architecture/timeouts.md

```diff
@@ -0,0 +1,92 @@
+
+```yaml
+plank: # a core component of Prow
+  default_decoration_configs:
+    '*':
+      grace_period: 30m0s
+      timeout: 4h0m0s
+    'org/repo': # overwrite the job timeout at repo level
+      grace_period: 45m0s
+      timeout: 6h0m0s
+```
+
+Any ephemeral cluster on AWS and GCE will be destroyed by [the periodic-ipi-deprovision job](https://github.com/openshift/release/blob/c3a73ae6c1161e242aeaa0943676827161936db8/ci-operator/jobs/infra-periodics.yaml#L377) after 4.5 hours since its creation. This job is to avoid leaking cloud resources in case that jobs fail to clean them up for any reason. **Note** that the `4.5h` here is globe and cannot be overwritten. It is not recommended to modify the job timeout if it impacts a job that requires an ephemeral cluster on those cloud providers.
+
+After Prow decides to delete the pod, `grace_period` is the time given to `ci-operator` before it is terminated by force. Its value is also specified in Prow's config, similarly to `timeout`.
+
+If `ci-operator` is still running after `grace_period`, the job log contains:
+
+```
+Process did not exit before 30m0s grace period
+```
+
+## Acquiring lease timeout: 2h
+
+Before installing the ephemeral cluster, `ci-operator` [acquires a lease of resource from Boskos](/docs/architecture/quota-and-leases/). The job log contains the following line if the acquisition has not succeeded after 2 hours:
+
+```
+... failed to acquire lease ... context deadline exceeded ...
+```
+
+The value `2h` is the default value of a flag to `ci-operator`. It cannot be changed for the jobs generated from ci-oerpator's config.
+
+## Step timeout
+
+Each [step](/docs/architecture/step-registry/#step) of a job runs in a pod. A step's `active_deadline_seconds` is passed directly through to the step's Pod.
+If the pod is still running after `active_deadline_seconds` since its creation, Kubernetes will mark it failed and delete it (see [upstream document](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#podspec-v1-core)). `ci-operator` checks the pod's status after it is complete and detects if the failure is due to `DeadlineExceeded`. In that case, the job log contains:
+
+```
+... steps failed ... exceeded the configured timeout ...
+```
+
+Similarly, `terminationGracePeriodSeconds` configured in a step is as well passed directly through to the step's Pod. It is the duration the pod needs to terminate gracefully (see [upstream document](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#podspec-v1-core)).
+
+The following step definition shows how to specify those values:
+
+```yaml
+ref:
+  as: org-repo-e2e
+  from: repo-tests
+  commands: org-repo-e2e-commands.sh
+  resources:
+    requests:
+      cpu: 1000m
+      memory: 100Mi
+  active_deadline_seconds: 600
+  terminationGracePeriodSeconds: 30
```

Comment:
Do we really not use `temination_grace_period_seconds` here? We should change it to be consistent.

Reply:
My mistake. Updated.